### PR TITLE
Add functionality to manipulate juju storage

### PR DESCRIPTION
**Pull request by lmlg**
This PR adds the needed functionality to add, detach and remove storage to juju models.

### CODE REVIEW

File: zaza/model.py [hunk 1]
```diff
@@ -2870,3 +2870,76 @@ async def async_get_cloud_data(credential_name=None, model_name=None):
...
+    :param pool: The pool on which to allocate the storage (i.e: cinder)
+    :type pool: str
+
+    :param size: The size in GB of the storage to attach.
+    :type size: int
+
+    :param model: The model name, or None, for the current model.
+    :type model: Option[str]
+
+    :returns: The name of the allocated storage.
```
Comment (ajkavanagh) [inline zaza/model.py:2893]: Needs a rtype here please (probably `str`?)

**Comment by ChrisMacNaughton** [comment]
As you (Alex) and I just discovered / discussed, the various storage methods [aren't actually implemented yet](https://github.com/juju/python-libjuju/blob/d67fa18c158a3980a048ee2bbc222ce19f44f1a5/juju/unit.py#L80)

**Comment by ajkavanagh** [comment]
> As you (Alex) and I just discovered / discussed, the various storage methods [aren't actually implemented yet](https://github.com/juju/python-libjuju/blob/d67fa18c158a3980a048ee2bbc222ce19f44f1a5/juju/unit.py#L80)

Yes; how incredibly annoying that the API docs call out the functions, with parameters and everything, but the implementations don't actually exist.

**Review by ChrisMacNaughton — APPROVED** [review]
I'd love to see a bug against libjuju to add this support there, but in the meantime, +1

**Comment by ChrisMacNaughton** [comment]
https://github.com/juju/python-libjuju/issues/638

**Pull request merged**